sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -174,6 +175,18 @@ private[sql] class HiveSessionCatalog(
   // xpath_double, xpath_float, xpath_int, xpath_long, xpath_number,
   // xpath_short, and xpath_string.
   override def lookupFunction(name: FunctionIdentifier, children: Seq[Expression]): Expression = {
+    try {
+      subLookupFunction(name, children)
+    } catch {
```

Comment:
@dongjoon-hyun Nevermind. We use the datatypes of the arguments passed to the HiveUDF/UDAF/UDFT to determine which object inspectors to use for conversion. So there is no way we can fix this using `ExpectsInputTypes`; sorry about the confusion...

We have only changed the default datatype for decimal conversion, so your I guess your fix is ok.